packages/cdktf-cli/src/bin/cmds/helper/performance.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import pidtree from "pidtree";
+import pidusage from "pidusage";
+
+async function getCombinedMemoryUsageInBytes(
+  pid = process.pid
+): Promise<number> {
+  const allPids = await pidtree(pid, { root: true });
+  const usages = await pidusage(allPids);
+  return Object.values(usages).reduce(
+    (carry, item) => (item && "memory" in item ? carry + item.memory : carry),
+    0
+  );
+}
+
+export function startPerformanceMonitoring() {
+  const start = Date.now();
```

Comment:
Nit: Should we continue to use `performance.now()` as we were doing [here](https://github.com/hashicorp/terraform-cdk/pull/2914/files#diff-711f3bf442c54e74867a9fa55969ab555b04f1bd03bd21b511bcddae7f805136L15). I do understand that the timelines we're interested in are much longer than performance.now() was designed for, so feel free to ignore my comment.

Reply:
Yeah I was thinking about that as well and chose Date.now since we don't need the fine grained resolution and it's in ms so a bit easier to calculate to seconds there :)